fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/container/SimpleFluidContainer.java

```diff
@@ -0,0 +1,113 @@
+package net.fabricmc.fabric.api.fluids.v1.container;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import com.google.common.collect.Iterables;
+
+import net.minecraft.fluid.Fluid;
+
+import net.fabricmc.fabric.Action;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.ImmutableFluidVolume;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.SimpleFluidVolume;
+
+public class SimpleFluidContainer implements FluidContainer {
+	private final Iterable<FluidContainer> containers;
+
+	public SimpleFluidContainer(FluidContainer... containers) {
+		this(Arrays.asList(containers));
+	}
+
+	public SimpleFluidContainer(Iterable<FluidContainer> containers) {
+		this.containers = containers;
+	}
+
+	@Override
+	public FluidVolume drain(Fluid fluid, long amount, Action simulate) {
+		FluidVolume current = new SimpleFluidVolume();
+
+		for (FluidContainer container : this.containers) {
+			FluidVolume drained = container.drain(current.isEmpty() ? fluid : current.getFluid(), amount, Action.SIMULATE);
+			if (!drained.equals(ImmutableFluidVolume.EMPTY)) {
+				current.consume(drained, Action.PERFORM);
+				container.drain(current.isEmpty() ? fluid : current.getFluid(), amount, simulate);
+			}
+			amount -= drained.getAmount();
+
+			if (amount <= 0) {
+				break;
+			}
+		}
+
+		return current;
+	}
+
+	@Override
+	public FluidVolume consume(FluidVolume volume, Action action) {
+		for (FluidContainer container : this.containers) {
+			if (volume.isEmpty()) break;
+			container.consume(volume, action);
+		}
+		return volume;
+	}
+
+	@Override
+	public Iterator<FluidVolume> iterator() {
+		return Iterables.concat(this.containers).iterator();
+	}
+
+	@Override
+	public boolean isEmpty() {
+		for (FluidContainer container : this.containers) {
+			if (!container.isEmpty()) {
```

Comment:
`stream().noneMatch(Container::isEmpty)`

Reply:
Cleaner, but you've used 'for' loops everywhere so this is fie too.

Reply:
Looking at it, there is a lot of places where streams would be extremely useful in compacting the code, but I will refrain from commenting there.

Reply:
for loop good

Reply:
stream best in game and cleaner

Reply:
![image](https://user-images.githubusercontent.com/25832468/83694155-d6de6800-a5cd-11ea-92d8-f4960186a7bf.png)

ur good nevermind, stream bad and worst in game

Reply:
yes